cmds/plugins/plugins.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package plugins
+
+import (
+	"errors"
+
+	"github.com/facebookincubator/contest/pkg/pluginregistry"
+
+	"github.com/facebookincubator/contest/pkg/job"
+	"github.com/facebookincubator/contest/pkg/target"
+	"github.com/facebookincubator/contest/pkg/test"
+
+	"github.com/facebookincubator/contest/plugins/reporters/noop"
+	"github.com/facebookincubator/contest/plugins/reporters/targetsuccess"
+	"github.com/facebookincubator/contest/plugins/targetmanagers/csvtargetmanager"
+	"github.com/facebookincubator/contest/plugins/targetmanagers/targetlist"
+	"github.com/facebookincubator/contest/plugins/testfetchers/literal"
+	"github.com/facebookincubator/contest/plugins/testfetchers/uri"
+	"github.com/facebookincubator/contest/plugins/teststeps/cmd"
+	"github.com/facebookincubator/contest/plugins/teststeps/echo"
+	"github.com/facebookincubator/contest/plugins/teststeps/example"
+	"github.com/facebookincubator/contest/plugins/teststeps/randecho"
+	"github.com/facebookincubator/contest/plugins/teststeps/slowecho"
+	"github.com/facebookincubator/contest/plugins/teststeps/sshcmd"
+	"github.com/sirupsen/logrus"
+)
+
+var targetManagers = []target.TargetManagerLoader{
+	csvtargetmanager.Load,
+	targetlist.Load,
+}
+
+var testFetchers = []test.TestFetcherLoader{
+	uri.Load,
+	literal.Load,
+}
+
+var testSteps = []test.TestStepLoader{
+	echo.Load,
+	slowecho.Load,
+	example.Load,
+	cmd.Load,
+	sshcmd.Load,
+	randecho.Load,
+}
+
+var reporters = []job.ReporterLoader{
+	targetsuccess.Load,
+	noop.Load,
+}
+
+// user-defined functions that will be made available to plugins for advanced
+// expressions in config parameters.
+var userFunctions = map[string]interface{}{
+	// sample function to prove that function registration works.
+	"do_nothing": func(a ...string) (string, error) {
+		if len(a) == 0 {
+			return "", errors.New("do_nothing: no arg specified")
+		}
+		return a[0], nil
+	},
+}
```

Comment:
Could you provide an example where these `userFunctions` are required? Why not to use a function directly? Is it for templates?